### ISSUE TITLE
v2: Support rolling over "in progress" todos

### ISSUE THREAD

**Issue by talios** [lumoe/obsidian-rollover-daily-todos#87]
It's becoming more common to have "in progress" todos, with the main Obsidian plugins now supporting custom emoji around these states as well, such as:

```
- [/] This is in progress
- [>] This is deferred and not currently being worked on, but wants to be tracked.
- [?] This is up for discussion, or needs more research
```

Currently, the roll over plugin seems to only move todo items with `[ ]` - incomplete, unfortunately this means that such "in progress" todos don't get rolled over.

It may be better if the plugin rolled over everything that's NOT `[x]` instead.

**Comment by tim-hilt** [comment]
Already work in progress: https://github.com/lumoe/obsidian-rollover-daily-todos/pull/76

**Comment by hilsonp** [comment]
I would really love to see this being implemented.
Indeed, I systematically tick the started tasks with a [/]  so that at the next stand-up meting, I know what I've been working on even though it is not finished yet.
Those unfinished should definitely be rolled over.
Another way is to let people modify the regex to suit their needs.

**Comment by hilsonp** [comment]
As a work-around, I modified the regex to allow forwarding the todo listed by @talios 
```
  // Returns true if string s is a todo-item
  #isTodo(s) {
    const r = /\s*- \[[ ?>\/]\].*/g;   //PHI: Patch to get unfinished, postponed and question items rolled over
    return r.test(s);
  }
```

**Comment by tim-hilt** [comment]
@hilsonp there is already a PR open for this: https://github.com/lumoe/obsidian-rollover-daily-todos/pull/76 although it’s vastly different from what @M31nevel requested.

On another note: the v2 release should primary be a stable, more maintainable rewrite. It’s not meant to provide a lot of new features. I would encourage you guys to discuss feature-requests in separate issues.

**Comment by MHolkamp** [comment]
> As a work-around, I modified the regex to allow forwarding the todo listed by @talios
> 
> ```
>   // Returns true if string s is a todo-item
>   #isTodo(s) {
>     const r = /\s*- \[[ ?>\/]\].*/g;   //PHI: Patch to get unfinished, postponed and question items rolled over
>     return r.test(s);
>   }
> ```

Works great!
Let's hop V2.0 will be live soon.
I can test if some one has a compiled version ;-)

**Comment by tim-hilt** [comment]
You need Node.js and pnpm to compile and run it.

Just click on the apple-icon on this page: https://nodejs.org/en/download/ to download a MacOS installer for Node.js. Then open up a terminal and run:

``` sh
npm install -g pnpm
```

to install pnpm, which is the package-manager that we use.

Uninstall Obsidian Rollover TODOs plugin from Obsidian. After that, navigate to `<your-vault-dir>/.obsidian/plugins` inside of a terminal. paste the unzipped content of the release-branch in a new folder `obsidian-rollover-todos` there (you can get the zip here: https://github.com/lumoe/obsidian-rollover-daily-todos/archive/refs/heads/release-2.0.zip) or `git clone` it there if you know how to use git.

After that run

``` sh
cd obsidian-rollover-todos &&pnpm i && pnpm build
```

the terminal will output a bunch of stuff and you should see a file called `main.json` in the same directory then.

You can know go to Obsidians settings on the community-plugins-page and you should see, that the Obsidian Rollover TODOs plugin is discovered again, but disabled. You can enable the plugin and take a look at the new settings.

**Comment by MHolkamp** [comment]
Awsome!!!

I know a bit of git ;-)
I tried the URL in the repository but then I get the old version.
Can you give me the url to the v2 version?
Then I can keep up with the newer beta version of v2.

Compiling is a matter of seconds.

-- 
-- 
Met vriendelijke groeten / Best regards / Mit freundlichen Grüße
Martin Holkamp-de Bruin

We hebben de wereld niet geërfd van onze ouders, maar te leen van onze kinderen,
We do not inherit the Earth from our parents; we borrow it from our Children.

Op 28 februari 2023 bij 17:39:07, Tim Hilt ***@***.***) schreef:

You need Node.js and pnpm to compile and run it.

Just click on the apple-icon on this page: https://nodejs.org/en/download/ to download a MacOS installer for Node.js. Then open up a terminal and run:

npm install -g pnpm
to install pnpm, which is the package-manager that we use.

Uninstall Obsidian Rollover TODOs plugin from Obsidian. After that, navigate to <your-vault-dir>/.obsidian/plugins inside of a terminal. paste the unzipped content of the release-branch in a new folder obsidian-rollover-todos there (you can get the zip here: https://github.com/lumoe/obsidian-rollover-daily-todos/archive/refs/heads/release-2.0.zip) or git clone it there if you know how to use git.

After that run

cd obsidian-rollover-todos &&pnpm i && pnpm build
the terminal will output a bunch of stuff and you should see a file called main.json in the same directory then.

You can know go to Obsidians settings on the community-plugins-page and you should see, that the Obsidian Rollover TODOs plugin is discovered again, but disabled. You can enable the plugin and take a look at the new settings.

**Comment by tim-hilt** [comment]
If you git cloned the whole repository, you can `cd` into the repo-directory and `git checkout release-2.0`

**Comment by MHolkamp** [comment]
Thanks!

Works great!


-- 
Met vriendelijke groeten / Best regards / Mit freundlichen Grüße
Martin Holkamp-de Bruin

We hebben de wereld niet geërfd van onze ouders, maar te leen van onze kinderen,
We do not inherit the Earth from our parents; we borrow it from our Children.

Op 1 maart 2023 bij 08:42:53, Tim Hilt ***@***.***) schreef:

pnpm i && pnpm build

**Comment by szpasztor** [comment]
@tim-hilt Thanks for the instructions. I followed them and every step works fine, except the last: "You can enable the plugin and take a look at the new settings."

The build succeeds, but for some reason I'm still getting no options: 
<img width="1112" alt="image" src="https://user-images.githubusercontent.com/8398290/235287327-2eb92dbd-e838-46c5-8684-4bc726b99b12.png">

Any idea why?

**Comment by tim-hilt** [comment]
@szpasztor is this related to #87? Otherwise, please comment on the correct issue. Otherwise I’m missing a lot of context

**Comment by szpasztor** [comment]
@tim-hilt Yes, I was referring to your comment on [Feb 28](https://github.com/lumoe/obsidian-rollover-daily-todos/issues/87#issuecomment-1448497476). I had followed the instructions there to get the "Support rolling over in progress todos" functionality. 

Based on your last note ("You can enable the plugin and take a look at the new settings."), my understanding is that a new plugin setting should appear in the settings panel, to enable that functionality. I tried that, but the settings are unchanged. 

Did I miss something here?

**Comment by tim-hilt** [comment]
Sounds like you’re hit by https://github.com/lumoe/obsidian-rollover-daily-todos/issues/93! Can you take a look at the comments and see if they are helpful? I might have forgetten to port my fix for #93 to the v2 branch

**Comment by edgarcosta** [comment]
Todos are not automatically rolled over with daily notes, I need to call: `Rollover daily todos: Rollover todos now` to roll them over.
